monitoring/opentelemetry/src/test/java/io/quarkus/ts/opentelemetry/OpenTelemetryProxyIT.java

```diff
@@ -0,0 +1,160 @@
+    private static final String BASIC_AUTH_PREFIX = "Basic ";
+    private static final String PROXY_USERNAME = "otel-user";
+    private static final String PROXY_PASSWORD = "otel-pwd";
+    private static final String JAEGER_MISSING_COLLECTOR = "missing.collector";
+
+    @JaegerContainer
+    static final JaegerService jaeger = new JaegerService();
+
+    @QuarkusApplication
+    static final RestService app = new RestService()
+            .withProperty("quarkus.otel.exporter.otlp.traces.proxy-options.enabled", "true")
+            .withProperty("quarkus.otel.exporter.otlp.traces.proxy-options.username", PROXY_USERNAME)
+            .withProperty("quarkus.otel.exporter.otlp.traces.proxy-options.password", PROXY_PASSWORD)
+            .withProperty("quarkus.otel.exporter.otlp.traces.proxy-options.port", OpenTelemetryProxyIT::getProxyPortAsString)
+            .withProperty("quarkus.otel.exporter.otlp.traces.proxy-options.host", "localhost")
+            .withProperty("quarkus.otel.exporter.otlp.traces.endpoint", "http://" + JAEGER_MISSING_COLLECTOR);
+
+    @Test
+    public void testProxyWithPassword(Vertx vertx) throws IOException {
+        try (var ignored = createGrpcProxy(vertx)) {
+            testTraces();
+        }
+    }
+
+    private static Closeable createGrpcProxy(Vertx vertx) {
+        GrpcServer grpcProxy = GrpcServer.server(vertx);
+
+        HttpServer proxyHttpServer = vertx.createHttpServer(new HttpServerOptions().setPort(getProxyPort()));
+        proxyHttpServer
+                .requestHandler(httpServerRequest -> {
+                    assertEquals(JAEGER_MISSING_COLLECTOR, httpServerRequest.authority().host());
+                    grpcProxy.handle(httpServerRequest);
+                })
+                .listen();
+
+        GrpcClient proxyClient = GrpcClient.client(vertx);
+
+        // create proxy server
+        // on request use gRPC client and pass the message to the Jaeger
+        grpcProxy.callHandler(reqFromQuarkus -> reqFromQuarkus.messageHandler(msgFromQuarkus -> proxyClient
+                .request(getJaegerSocketAddress())
+                .onSuccess(requestToJaeger -> {
+                    assertProxyUsernameAndPassword(reqFromQuarkus);
+                    requestToJaeger
+                            .methodName(reqFromQuarkus.methodName())
+                            .serviceName(reqFromQuarkus.serviceName())
+                            .end(msgFromQuarkus.payload());
+                    // send Jaeger response back to the Quarkus application
+                    requestToJaeger.response()
+                            .onSuccess(h -> h.messageHandler(msg -> reqFromQuarkus.response().endMessage(msg)))
+                            .onFailure(err -> reqFromQuarkus.response().status(GrpcStatus.ABORTED).end());
+                })));
+        return () -> CompletableFuture.allOf(proxyHttpServer.close().toCompletionStage().toCompletableFuture(),
```

Comment:
is not needed add some CountDownLatch to help the concurrency management? I looked something similar here--> https://github.com/eclipse-vertx/vertx-http-proxy/blob/0d7bad50c138fee43baf6f3506d5ebd4ee01ddc5/src/test/java/io/vertx/httpproxy/TestBase.java#L94C11-L94C25

Reply:
if you mean that gRPC client should be closed before HTTP server, I think you would be technically right, not sure it matters, but I'll look into it.

Reply:
I have chained these 2 operations. I don't have too much of experience with this, hopefully it's done correctly.